### ISSUE TITLE
dotenv-path loads from invocation directory

### ISSUE THREAD

**Issue by stevenh** [casey/just#1927]
I added a `set dotenv-path := "build/.env"` to a Justfile expecting that it would process the same as any other component of the targets i.e. it would implicitly load from `justfile_directory()` e.g. it would load `justfile_directory()/build/.env` however this is not the case instead it loads from  the invocation directory so only works it the users working directory is the same as `justfile_directory()`.

Next I tried explicitly specifying to use the `Justfile` directory using the following but also no good:
```Justfile
set dotenv-path := "{{justfile_directory()}}/build/.env"
```

Finally I tried switching to use `dotenv-filename` which did what I wanted and now loads the environment file from the path relative to the `Justfile`

Workaround:
```
set dotenv-filename := "build/.env"
```

It's not clear to me [from the documentation](https://github.com/casey/just?tab=readme-ov-file#table-of-settings) if I'm miss-understanding the intended behaviour different between `dotenv-filename` and `dotenv-path` or if `dotenv-path` is not working as intended. However I would have expected the fully qualified version using `"{{justfile_directory()}}/build/.env"` to work correctly.

```shell
just --version
just 1.24.0
```

**Comment by stevenh** [comment]
This [PR seems related](https://github.com/casey/just/pull/661), but possibly only fixed `dotenv-filename`?

**Comment by casey** [comment]
Thanks for the report! This is definitely a bug. `dotenv-path` should be interpreted relative to the working directory, not the invocation directory. Fixed in #2040.
